ndk-build/src/apk.rs

```diff
@@ -84,8 +84,15 @@ impl<'a> UnalignedApk<'a> {
         std::fs::create_dir_all(out.parent().unwrap())?;
         std::fs::copy(path, out)?;
 
+        // Pass UNIX path separators to `aapt` on non-UNIX systems, ensuring the resulting separator
+        // is compatible with the target device instead of the host platform.
+        // Otherwise, it results in a runtime error when loading the NativeActivity `.so` library.
```

Comment:
I guess this is just a ``"failed to find/load `lib/<target arch>/lib<activityname>.so, -ENOENT`"`` kind of error?

Reply:
yes, but not really clear imo why it fails:

AndroidRuntime: java.lang.RuntimeException: Unable to start activity ComponentInfo{rust.wgpu_android/android.app.NativeActivity}: java.lang.IllegalArgumentException: Unable to find native library wgpu_android using classloader: dalvik.system.PathClassLoader[DexPathList[[],nativeLibraryDirectories=[/data/app/rust.wgpu_android-pSe8ljSIP942rgeN5FSlQQ==/lib/arm64, /system/lib64, /system/product/lib64]]]

Reply:
(That's exactly the kind of context we should paste in PR's / commit messages from the beginning :wink::wink:. Both for reviewer-understanding as well as users running into this issue to have a higher chance finding it with the search bar)

Seems pretty clear to me: try running `touch foo\\bar\\baz` on a UNIX system. This creates a file with that name, instead of the directory structure. Same happens here, see https://github.com/rust-windowing/android-ndk-rs/runs/3165210588:

```
Verifying alignment of D:\a\android-ndk-rs\android-ndk-rs\target\debug\apk\examples\hello_world.apk (4)...
      49 AndroidManifest.xml (OK - compressed)
     880 lib\arm64-v8a\libhello_world.so (OK - compressed)
```

`lib\arm64-v8a\libhello_world.so` is a filename in the root of the zip archive - it's not a directory structure. This doesn't get unpacked to `/data/app/rust.wgpu_android-pSe8ljSIP942rgeN5FSlQQ==/lib/arm64` so that the class loader finally fails to find the library.